CDH2.py

```diff
@@ -0,0 +1,348 @@
+import os
+from Bio import SeqIO
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+from Bio.SeqUtils import gc_fraction
+from typing import Union, Tuple, Dict
+from abc import ABC, abstractmethod
+
+
+
+def analyse_gc(records: str, min_gc: Union[int, float], max_gc: Union[int, float]) -> str:
```

Comment:
Можно было бы более точно отразить суть функции в названии и скорректировать аннотацию типов:
```suggestion
def filter_by_gc_content(records: List[SeqRecord], min_gc: Union[int, float], max_gc: Union[int, float]) -> Dict[str, SeqRecord]:
```
Идея возвращать словарь видимо была навеяна предыдущей реализацией fastq-фильтратора, но можно сделать и проще: возвращать список из объектов SeqRecord, там  уже есть вся информация о последовательнотях в атрибутах. Ты здесь пытаешься добавить дополнительную структуру, используя словарь, но она ведь уже есть :)